core/arch/arm/plat-imx/mmdc.c

```diff
@@ -17,17 +17,11 @@
 
 int imx_get_ddr_type(void)
 {
-	uint32_t val, off;
 	bool is_mx7 = soc_is_imx7ds();
-	vaddr_t mmdc_base;
-
-	if (is_mx7)
-		off = DDRC_MSTR;
-	else
-		off = MMDC_MDMISC;
-
-	mmdc_base = core_mmu_get_va(MMDC_P0_BASE, MEM_AREA_IO_SEC,
-				    off + sizeof(uint32_t));
+	uint32_t val;
```

Comment:
`	uint32_t val = 0;`

Reply:
Now val and off are left as they were before my commit, so I wondering - is it ok now?

Reply:
A few years ago we changed the coding guidelines to always initialize all local variables with some value. Not all functions follow this convention. When we make changes to such functions we often take care of this initialization too, especially if we make changes to local variables. So please initialize all the currently uninitialized locals with `0`, preferably with not more than one variable per line.